### PR TITLE
New config

### PR DESCRIPTION
**Pull request by PaLeroy**
WiP PR for the new config files.
I suggest we add here the best config for uc_10 and uc_50 as a first start.

### CODE REVIEW

File: pymarl/config/algs/qmix.yaml [hunk 1]
```diff
@@ -4,14 +4,14 @@
 action_selector: "epsilon_greedy"
 epsilon_start: 1.0
 epsilon_finish: 0.05
-epsilon_anneal_time: 200000
+epsilon_anneal_time: 5000
```
Comment (moratodpg) [inline pymarl/config/algs/qmix.yaml:7]: Shall we do the same for algorithms as you have done for environment configuration files?
I.e., Perhaps this configuration of qmix is great for n=10 (and n=50), but could we also add a different one for n=100?

File: pymarl/modules/critics/coma.py [hunk 2]
```diff
@@ -14,9 +14,10 @@ def __init__(self, scheme, args):
         input_shape = self._get_input_shape(scheme)
         self.output_type = "q"
 
+        critic_size = self.args.critic_intermediate_size
         # Set up network layers
         self.fc1 = nn.Linear(input_shape, 128)
-        self.fc2 = nn.Linear(128, 128)
+        self.fc2 = nn.Linear(critic_size, 128)
```
Comment (moratodpg) [inline pymarl/modules/critics/coma.py]: Shouldn't we also change 128 as the output size of both fc1 and fc2 to critic_size?

**Review by moratodpg — COMMENTED** [review]
I have checked it! Awesome idea to improve reproducibility.
I just added two minor comments and then we can of course merge.

File: pymarl/config/algs/qmix.yaml (hunk 1 above)
Comment (PaLeroy) [reply pymarl/config/algs/qmix.yaml:7]: Yes

File: pymarl/config/envs/struct_c_10.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  n_comp: 10
+  discount_reward: 0.95
+  k_comp: 9
+  state_obs: True
+  state_d_rate: True
+  state_alphas: True
+  obs_d_rate: False
+  obs_multiple: False
+  obs_all_d_rate: False
+  obs_alphas: False
```
Comment (PaLeroy) [inline pymarl/config/envs/struct_c_10.yaml]: Do we add obs_alphas for the correlated env?

Comment (moratodpg) [reply pymarl/config/envs/struct_c_10.yaml]: In my previous experiments, I see that adding alphas improves a lot the results.

Comment (PaLeroy) [reply pymarl/config/envs/struct_c_10.yaml]: let's do it then.
It scales linearly with the number of agents and we are supposed to be in a perfect situation information

Comment (PaLeroy) [reply pymarl/config/envs/struct_c_10.yaml]: Done in [2aa7a1f](https://github.com/moratodpg/marl_struct/pull/16/commits/2aa7a1f55b0e3d61373594f5f4c16ffb2aec85d6)

File: pymarl/config/algs/qvmix_uc_100.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+# use epsilon greedy action selector
+action_selector: "epsilon_greedy"
+epsilon_start: 1.0
+epsilon_finish: 0.05
+epsilon_anneal_time: 100000
+
+runner: "episode"
+#batch_size_run: 6
+
+batch_size: 32
```
Comment (PaLeroy) [inline pymarl/config/algs/qvmix_uc_100.yaml]: Did you test with batch_size 64 as the others?

Comment (moratodpg) [reply pymarl/config/algs/qvmix_uc_100.yaml]: Not yet. I will do it.

Comment (moratodpg) [reply pymarl/config/algs/qvmix_uc_100.yaml]: I just changed it now.

File: pymarl/config/algs/qvmix_uc_50.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+action_selector: "epsilon_greedy"
+epsilon_start: 1.0
+epsilon_finish: 0.05
+epsilon_anneal_time: 200000
+
+runner: "episode"
+
+batch_size: 64
+buffer_size: 5000
```
Comment (moratodpg) [inline pymarl/config/algs/qvmix_uc_50.yaml]: Is this correct?

File: pymarl/config/algs/comaIS.yaml [hunk 6]
```diff
@@ -32,3 +32,6 @@ q_nstep: 0  # 0 corresponds to default Q, 1 is r + gamma*Q, etc
 
 mac: "is_mac"
 name: "comaIS"
+
+use_cuda: False
```
Comment (PaLeroy) [inline pymarl/config/algs/comaIS.yaml:36]: What about coma_is? :p

File: pymarl/config/algs/iql.yaml [hunk 7]
```diff
@@ -4,17 +4,14 @@
 action_selector: "epsilon_greedy"
```
Comment (PaLeroy) [inline pymarl/config/algs/iql.yaml:3]: iql still needs to be checked

File: pymarl/config/algs/qmix_uc_100.yaml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+batch_size: 64
+buffer_size: 5000
+target_update_interval: 200
+
+agent_output_type: "q"
+learner: "q_learner"
+double_q: True
+mixer: "qmix"
+mixing_embed_dim: 64
```
Comment (PaLeroy) [inline pymarl/config/algs/qmix_uc_100.yaml]: Not sure about the mixing embed size

Comment (PaLeroy) [reply pymarl/config/algs/qmix_uc_100.yaml]: (for uc100)

Comment (PaLeroy) [reply pymarl/config/algs/qmix_uc_100.yaml]: resolved

**Comment by PaLeroy** [comment]
Before merge: agree on DQN and IQL.
For me, they could be the same as QMix and QVMix in term of epsilon and buffer/batch sizes

File: pymarl/config/algs/comaIS.yaml (hunk 6 above)
Comment (moratodpg) [reply pymarl/config/algs/comaIS.yaml:36]: I am thinking of not considering it in our study. We might explore this at some point though. Thus we can just keep it as default, of course, if you agree.

File: pymarl/config/algs/qmix_uc_100.yaml [hunk 9]
```diff
@@ -15,7 +15,7 @@ agent_output_type: "q"
 learner: "q_learner"
 double_q: True
 mixer: "qmix"
-mixing_embed_dim: 64
+mixing_embed_dim: 128
```
Comment (moratodpg) [inline pymarl/config/algs/qmix_uc_100.yaml:18]: In my case, it works rather well with 64. You found better results with 128?

File: pymarl/config/algs/comaIS.yaml (hunk 6 above)
Comment (PaLeroy) [reply pymarl/config/algs/comaIS.yaml:36]: agreed

File: pymarl/config/algs/qmix_uc_100.yaml (hunk 9 above)
Comment (PaLeroy) [reply pymarl/config/algs/qmix_uc_100.yaml:18]: In average yes, but it is also to be consistent with qvmix

Comment (moratodpg) [reply pymarl/config/algs/qmix_uc_100.yaml:18]: Okay, then let's keep 128 :)

**Comment by PaLeroy** [comment]
Everything seems to be ok for n_comp < 50.
Just some tests to be sure for the 100 param.
+ Some modif for dqn to come to increase network capacity.
I think we can merge the PR for now?

**Review by moratodpg — APPROVED** [review]
I briefly went through the PR and have not found any issues. Let's merge it. :)

**Review by PaLeroy — COMMENTED** [review]
Let's go then 😄

**Pull request merged**